server/controllers/caseNotesController.ts

```diff
@@ -194,10 +185,17 @@ export default class CaseNotesController {
         hours,
         minutes,
       }
+
       const errors = req.errors || []
       if (!errors.length) {
+        const allowedCaseNoteTypes = await this.caseNotesService.getCaseNoteTypesForUser(userToken)
+        if (!allowedCaseNoteTypes.some(allowedType => allowedType.code === type)) {
+          logger.info(`User not permitted to create case note of type: ${type}`)
+          return next()
+        }
```

Comment:
We now need to take ownership of checking that the user can actually create a case note of the given type